### DISCUSSION TITLE
Lazy loading

### DISCUSSION THREAD

**Discussion by Valexr** [Valexr/Slidy#10] [Ideas]
What do you think about lazy loading images function in core?

**Comment by subpx** [comment]
As you mentioned in the Svelte discord, I think this should not be the job of Slidy. Some lifecycle events would be useful.

For a start: page-change, transition-start, transition-end

**Reply by Valexr** [reply]
Thnx for advice @subpx 👍🏻
What do you mean about this name of lifecycle, can you explain little bit more...?

**Reply by subpx** [reply]
**page-change** 
dispatched after a page change, and contains current page number

**transition-start**
dispatched at start of page transition

**transition-end**
dispatched at end of page transition

**Reply by Valexr** [reply]
What does a page mean?

**Reply by subpx** [reply]
page = slide

**Comment by Valexr** [comment]
I think maybe something like this https://svelte.dev/repl/9defdb17ba0942158798777944a0e1aa

**Reply by subpx** [reply]
Not sure what this is an example of?

**Reply by Valexr** [reply]
It’s my example with InterectionObserver to trigger slides in viewport🤓

**Comment by Valexr** [comment]
Lazy example https://svelte.dev/repl/289a4972eb23415a98da2d059aa5d766
